tests/contrib/hooks/test_gcp_pubsub_hook.py

```diff
@@ -0,0 +1,297 @@
+
+TEST_PROJECT = 'test-project'
+TEST_TOPIC = 'test-topic'
+TEST_SUBSCRIPTION = 'test-subscription'
+TEST_UUID = 'abc123-xzy789'
+TEST_MESSAGES = [
+    {
+        'data': b64e('Hello, World!'),
+        'attributes': {'type': 'greeting'}
+    },
+    {'data': b64e('Knock, knock')},
+    {'attributes': {'foo': ''}}]
+
+EXPANDED_TOPIC = 'projects/%s/topics/%s' % (TEST_PROJECT, TEST_TOPIC)
+EXPANDED_SUBSCRIPTION = 'projects/%s/subscriptions/%s' % (TEST_PROJECT,
+                                                          TEST_SUBSCRIPTION)
+
+
+def mock_init(self, gcp_conn_id, delegate_to=None):
+    pass
+
+
+class PubSubHookTest(unittest.TestCase):
+    def setUp(self):
+        with mock.patch(BASE_STRING.format('GoogleCloudBaseHook.__init__'),
+                        new=mock_init):
+            self.pubsub_hook = PubSubHook(gcp_conn_id='test')
+
+    @mock.patch(PUBSUB_STRING.format('PubSubHook.get_conn'))
+    def test_create_nonexistent_topic(self, mock_service):
+        self.pubsub_hook.create_topic(TEST_PROJECT, TEST_TOPIC)
+
+        create_method = (mock_service.return_value.projects.return_value.topics
+                         .return_value.create)
+        create_method.assert_called_with(body={}, name=EXPANDED_TOPIC)
+        create_method.return_value.execute.assert_called_with()
+
+    @mock.patch(PUBSUB_STRING.format('PubSubHook.get_conn'))
+    def test_delete_topic(self, mock_service):
+        self.pubsub_hook.delete_topic(TEST_PROJECT, TEST_TOPIC)
+
+        delete_method = (mock_service.return_value.projects.return_value.topics
+                         .return_value.delete)
+        delete_method.assert_called_with(topic=EXPANDED_TOPIC)
+        delete_method.return_value.execute.assert_called_with()
+
+    @mock.patch(PUBSUB_STRING.format('PubSubHook.get_conn'))
+    def test_delete_nonexisting_topic_failifnotexists(self, mock_service):
+        (mock_service.return_value.projects.return_value.topics
+         .return_value.delete.return_value.execute.side_effect) = HttpError(
+            resp={'status': '404'}, content='')
+
+        try:
+            self.pubsub_hook.delete_topic(TEST_PROJECT, TEST_TOPIC)
+            self.fail('Topic deletion should fail for non-existing topic when '
+                      'fail_if_not_exists=True')
+        except Exception:
+            pass  # Expected.
+
+    @mock.patch(PUBSUB_STRING.format('PubSubHook.get_conn'))
+    def test_create_preexisting_topic_failifexists(self, mock_service):
+        (mock_service.return_value.projects.return_value.topics.return_value
+         .create.return_value.execute.side_effect) = HttpError(
+            resp={'status': '409'}, content='')
+
+        try:
+            self.pubsub_hook.create_topic(TEST_PROJECT, TEST_TOPIC,
+                                          fail_if_exists=True)
+        except Exception:
+            pass  # Expected.
+        else:
+            self.fail('Topic creation should fail for existing topic when '
```

Comment:
nit: This style of failing on success is different from the previous test case's. Can you make the two consistent?